himbaechel/uarch/gatemate/gen/arch_gen.py

```diff
@@ -226,12 +222,30 @@ def main():
         print("==============================================================================")
         os._exit(-1)
 
+    new_wires = dict()
+    wire_delay = dict()
+    for _,nodes in dev.get_connections():
+        for conn in nodes:
+            if conn.endpoint:
+                t_name = dev.get_tile_type(conn.x,conn.y)
+                if t_name not in new_wires:
+                    new_wires[t_name] = set()
+                new_wires[t_name].add(conn.name)
+                # Check to confirm no duplicates
+                if conn.name in wire_delay:
+                    if wire_delay[conn.name]!=conn.delay:
+                        print(conn.name, conn.delay)
```

Comment:
I think this seems like it should be an assertion rather than a print at this point?